### PR TITLE
Lms/grammar rule data migration

### PR DESCRIPTION
**Pull request by happythenewsad**
## WHAT
a data migration for grammar rules

## WHY
As a developer, I want grammar feedback available in the activity report, so that I can analyze
(the Rules Report tool implicitly filters feedbacks not associated with Rules, via JOIN statements)

## HOW
rake task + csv file 

### Screenshots
(If applicable. Also, please censor any sensitive data)

### Notion Card Links
https://www.notion.so/quill/Enter-Grammar-Rules-into-the-comprehension_rules-DB-653daceea79a4297925d30d54b08cd46

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  no - simple rake task
Have you deployed to Staging? | not yet
Self-Review: Have you done an initial self-review of the code below on Github? | yes
Design Review: If applicable, have you compared the coded design to the mockups? | n/a

### CODE REVIEW

File: services/QuillLMS/lib/tasks/add_grammar_api_rules_and_feedback.rake
Comment (dandrabik) [inline services/QuillLMS/lib/tasks/add_grammar_api_rules_and_feedback.rake]: I believe we'd want the suborder to be the precedence from the grammar API, rather than a hard-coded value. (It's also zero-based).

File: services/QuillLMS/lib/tasks/add_grammar_api_rules_and_feedback.rake
Comment (dandrabik) [inline services/QuillLMS/lib/tasks/add_grammar_api_rules_and_feedback.rake]: Order is zero-based, so these should all be `0`.

File: services/QuillLMS/lib/data/rules.csv
Comment (dandrabik) [inline services/QuillLMS/lib/data/rules.csv]: This line seems like it needs to be removed.

File: services/QuillLMS/lib/tasks/add_grammar_api_rules_and_feedback.rake
Comment (anathomical) [inline services/QuillLMS/lib/tasks/add_grammar_api_rules_and_feedback.rake]: This isn't super critical, but I think we actually want to put something here if we have it.  It's not dead-on, but I suggest using `Overall Rule Priority`.  Alternatively, since it looks like the CSV is in priority order, maybe we could just use the `index + 1` of where we are in the `each` call.

File: services/QuillLMS/lib/tasks/add_grammar_api_rules_and_feedback.rake
Comment (anathomical) [inline services/QuillLMS/lib/tasks/add_grammar_api_rules_and_feedback.rake]: It doesn't super matter, but I think we probably want to populate this with `Feedback - Revised`

File: services/QuillLMS/lib/tasks/add_grammar_api_rules_and_feedback.rake
Comment (anathomical) [inline services/QuillLMS/lib/tasks/add_grammar_api_rules_and_feedback.rake]: I think we also want a loop that creates the `opinion` rules as well.

**Review by anathomical — CHANGES_REQUESTED** [review]
A couple of tweaks I think are worth making, but generally speaking this makes sense to me.

File: services/QuillLMS/lib/data/rules.csv
Comment (dandrabik) [inline services/QuillLMS/lib/data/rules.csv]: This looks like a generic 'Grammar API rule', should this be here?

File: services/QuillLMS/lib/data/rules.csv
Comment (dandrabik) [inline services/QuillLMS/lib/data/rules.csv]: Where did this CSV come from? Is this an export from Notion? There's a lot of non-grammar rule cruft in here.

File: services/QuillLMS/lib/tasks/add_grammar_api_rules_and_feedback.rake
Comment (dandrabik) [reply services/QuillLMS/lib/tasks/add_grammar_api_rules_and_feedback.rake]: Agreed that we'd want to populate this with the current feedback.

File: services/QuillLMS/lib/tasks/add_grammar_api_rules_and_feedback.rake
Comment (dandrabik) [inline services/QuillLMS/lib/tasks/add_grammar_api_rules_and_feedback.rake]: I believe these files (csv and rake task) should live in the Comprehension Engine?

File: services/QuillLMS/lib/data/rules.csv
Comment (dandrabik) [inline services/QuillLMS/lib/data/rules.csv]: Does this line have unbalanced quotes? Is that going to cause import/parsing issues?

**Review by dandrabik — CHANGES_REQUESTED** [review]
There are a few items that need to be cleaned up here, e.g. we should be populating the order and feedback.

Also, can you take a moment and fill out the PR description more. Please fill out the 'Why' as a communication tool to other developers. Also, the PR description is a good place to mention: Where did this `csv` come from? Why does it have so many non-grammar rules in it? Why did you decide to go with a placeholder  `1` for `suborder`? Why did you decide to use `'lorem ipsum'` instead of the feedback from the `csv`? etc.

Also, I'm a little worried about the approach here of starting out with an export file with a lot of cruft (which I think is from Notion) and then filtering it down in code. I assuming you took this approach to make it easy for the Curriculum Team. It might be worth maintain a google sheet with just the columns that are being imported. From a reviewer's perspective, it's hard to parse what will end up in the production DB. I flagged one rule which seems like a universal `Grammar API` Rule.  Is that supposed to be copied to the DB? Another line had unbalanced quotes, which also might cause parsing issues.

File: services/QuillLMS/lib/tasks/add_grammar_api_rules_and_feedback.rake
Comment (happythenewsad) [inline services/QuillLMS/lib/tasks/add_grammar_api_rules_and_feedback.rake]: Rubocop throws Layout/AssignmentIndentation: Indent the first line of the right-hand-side of a multi-line assignment.

even though the line seems indented correctly

**Comment by happythenewsad** [comment]
FYI @anathomical 
Dan and I just synced on this. We're holding off further review of this PR until I meet with Lindsey tomorrow to establish a plan for keeping the input data clean and up to date.

File: services/QuillLMS/lib/tasks/add_grammar_api_rules_and_feedback.rake
Comment (happythenewsad) [reply services/QuillLMS/lib/tasks/add_grammar_api_rules_and_feedback.rake]: Hannah created a new column for this today, which I am now using.

File: services/QuillLMS/lib/tasks/add_grammar_api_rules_and_feedback.rake
Comment (happythenewsad) [reply services/QuillLMS/lib/tasks/add_grammar_api_rules_and_feedback.rake]: Hannah explicitly added correct values in the suborder col today.

**Comment by happythenewsad** [comment]
The CSV file should be clean now. `Rule.suborder` is now extracted directly from the CSV. `Feedback.order` is set to `1` for all feedbacks created from this task. Also, the task is now idempotent.

File: services/QuillLMS/lib/data/rules.csv
Comment (happythenewsad) [reply services/QuillLMS/lib/data/rules.csv]: All rows were parsed correctly in the latest CSV (which is fresher than this comment)

File: services/QuillLMS/engines/comprehension/lib/tasks/add_grammar_api_rules_and_feedback.rake
Comment (dandrabik) [inline services/QuillLMS/engines/comprehension/lib/tasks/add_grammar_api_rules_and_feedback.rake]: This should be 0, since this is zero-based.

File: services/QuillLMS/engines/comprehension/lib/tasks/add_grammar_api_rules_and_feedback.rake
Comment (dandrabik) [inline services/QuillLMS/engines/comprehension/lib/tasks/add_grammar_api_rules_and_feedback.rake:18]: Flagging that the order is zero-based. I think the orders in the csv start at 1.

**Review by dandrabik — APPROVED** [review]
Thanks for making these changes!

So approving if a small change is made: changing the order and suborder to zero-based (instead of starting at 1).

File: services/QuillLMS/engines/comprehension/lib/tasks/rules.csv
Comment (dandrabik) [inline services/QuillLMS/engines/comprehension/lib/tasks/rules.csv]: The suborder seems to start at 1, and should start at zero.

**Review by anathomical — APPROVED** [review]

**Pull request merged**
